bee-network/src/peer/list.rs

```diff
@@ -0,0 +1,456 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{error::Error, meta::PeerState};
+
+use crate::{
+    alias,
+    config::Peer,
+    init::global::max_unknown_peers,
+    swarm::protocols::gossip::GossipSender,
+    types::{PeerInfo, PeerRelation},
+};
+
+use libp2p::{Multiaddr, PeerId};
+use tokio::sync::RwLock;
+
+use std::{
+    collections::{HashMap, HashSet},
```

Comment:
Use `hashbrown` if you can, `ahash` is much more performant.

Reply:
This is just a `PeerList` however, which doesn't even contain many peers. Do you still think it's worth?

Reply:
I switched to `hashbrown`. I don't think it will be noticable in this particular scenario, but faster is always better than slower, so why not. I think the extra compilation time for `hashbrown` is negligable.